applications/plugins/SofaPython/Binding_Base.cpp

```diff
@@ -37,13 +37,30 @@
 #include "PythonEnvironment.h"
 using sofa::simulation::PythonEnvironment ;
 using sofa::core::topology::BaseMeshTopology ;
-using sofa::helper::vector ;
-using namespace sofa::core::objectmodel;
+typedef BaseMeshTopology::Tetra Tetra;
+using sofa::helper::vector;
 using sofa::helper::Factory;
+using namespace sofa::core::objectmodel;
 
-
+// TODO (Stefan Escaida 13.02.2018): this factory code is redundant to the Communication plugin, but should easily be mergeable, when an adequate spot is found.
```

Comment:
Nice ! What did you plan for the redundant part ?

Reply:
Eventually, one and the same factory could be used for creating the Data for Sofa Communication and for the PSDE. Merging the code into one should be trivial. Yet, as far as I understand, currently there is no obvious place in the structure where this factory should be --> future work?

Reply:
Not sure than merging code into one is consistent. They are both doing something similar but it's a simple instanciation of a factory class.